src/Symfony/Component/Config/Resource/DirectoryResource.php

```diff
@@ -87,7 +87,12 @@ public function isFresh($timestamp)
                 continue;
             }
 
-            $newestMTime = max($file->getMTime(), $newestMTime);
+            // early return if any file mtime is greater than passed timestamp
+            if (($fileMTime = $file->getMTime()) > $timestamp) {
+                return false;
+            }
+
+            $newestMTime = max($fileMTime, $newestMTime);
```

Comment:
I too think that the same timestamp should be considered false. Can you clarify why the behavior exists such that a matching timestamp is *not fresh* @nicolas-grekas